### PR TITLE
feat: some fun unsafe stuff

### PR DESCRIPTION
**Pull request by Will-Smith11**
@Autoparallel  Remember at ethcc when we couldn't find a way around the PendingTransaction state and you had to end up forking ethers-rs...
Well I realized I had a fix and wanted to try this. I have a local test and the code works as expected... I don't care if you merge it or not I just wanted a low level challenge and thought best case scenario I can also help someone else out.

### CODE REVIEW

File: arbiter-core/src/middleware.rs [hunk 1]
```diff
@@ -180,21 +183,43 @@ impl Middleware for RevmMiddleware {
...
+                };
+
+                let faked_transaction = PendingTransactionMock::new(
+                    ethers::types::H256::zero(),
+                    self.provider(),
+                    block,
+                    tx_receipts,
+                );
+
+                unsafe { Ok(std::mem::transmute(faked_transaction)) }
```
Comment (Will-Smith11) [inline arbiter-core/src/middleware.rs:203]: this is more or less the only actually changes. rest done by my fmter and deps

**Comment by Will-Smith11** [comment]
https://github.com/Will-Smith11/mem-playground 
if you want to test it btw

**Comment by Autoparallel** [comment]
@Will-Smith11 can we meet today and talk about this? I'd like to understand how it works. Dropping our ethers fork would be hugely beneficial. 

I'll need one more feature than you list here, I believe. But nonetheless, this is awesome. I love to see it!!

**Comment by Autoparallel** [comment]
Just sent a PR into your branch @Will-Smith11 

I reduced things a bit. 

Let's keep it going and make sure we can get this to work with the least amount of overhead and with as much compatibility as possible.

**Comment by 0xJepsen** [comment]
Lets be sure to actually fill out all the useful fields in the transactions receipts

**Comment by 0xJepsen** [comment]
I am going to close this PR and Re-open it into the `feat(arbiter-core)/v0.5.0` branch as we will include this in the next minor version

**Pull request closed without merging**